### PR TITLE
Delete Comment Functionality

### LINKED ISSUE #123: users should be able to delete their own comments and admins should be able to delete all comments

### PR DESCRIPTION
**Pull request by sehgalvibhor**
Closes #123 

Super Admin View :
superAdmin

Full User View :
userComment

### CODE REVIEW

File: skyportal/handlers/comment.py [hunk 1]
```diff
@@ -133,10 +133,17 @@ def delete(self, comment_id):
...
+        user = (self.current_user.username if hasattr(self.current_user, 'username') else self.current_user.name)
+        roles = (self.current_user.roles if hasattr(self.current_user, 'roles') else '')
+        c = Comment.query.get(comment_id)
+        source_id = c.source_id
+        author = c.author
+        if ("Super admin" in [role.id for role in roles]) or (user == author):
+                try:
+                        Comment.query.filter_by(id=comment_id).delete()
+                        DBSession().commit()
+                except ValidationError as e:
```
Comment (sehgalvibhor) [inline skyportal/handlers/comment.py]: Not sure which Marshmallow validation to use here, need help!

**Comment by acrellin** [comment]
This looks great, thanks @sehgalvibhor!  Just needs a test - if you look in `skyportal/tests/frontend/test_sources.py` at `test_comments`, that should give you an idea of how to test this new feature. Simply adding another function below `test_comments` to test deleting would be great!

**Comment by sehgalvibhor** [comment]
Sure! I'll have a look and add the test.

**Comment by acrellin** [comment]
@sehgalvibhor just wanted to check in on the status of this PR - do you need a hand getting a test in place?

**Comment by sehgalvibhor** [comment]
@acrellin : Completely missed this, I should be able tot add the tests (and resolve conflicts) over the weekend. Thanks!

**Comment by acrellin** [comment]
Moved to https://github.com/skyportal/skyportal/pull/254

**Pull request closed without merging**